parsel/selector.py

```diff
@@ -35,12 +36,24 @@ def _st(st):
         raise ValueError('Invalid type: %s' % st)
 
 
-def create_root_node(text, parser_cls, base_url=None):
+def create_root_node(text, parser_cls, base_url=None, huge_tree=True):
     """Create root node for text using given parser class.
     """
+    lxml_version = parse_version(etree.__version__)
+    lxml_huge_tree_version = parse_version("4.2")
+
     body = text.strip().encode('utf8') or b'<html/>'
-    parser = parser_cls(recover=True, encoding='utf8')
-    root = etree.fromstring(body, parser=parser, base_url=base_url)
+    if huge_tree and lxml_version >= lxml_huge_tree_version:
+        parser = parser_cls(recover=True, encoding='utf8', huge_tree=True)
+        root = etree.fromstring(body, parser=parser, base_url=base_url)
+    else:
+        parser = parser_cls(recover=True, encoding='utf8')
+        root = etree.fromstring(body, parser=parser, base_url=base_url)
+        for error in parser.error_log:
+            if 'use XML_PARSE_HUGE option' in error.message:
+                raise ValueError("input data too big. Upgrade to lxml %s for huge_tree support." % (
```

Comment:
Won't this exception be confusing with huge_tree=False passed explicitly?

Reply:
Well, yes. The question is what to expect when parsing a huge document, when the program doesn't support it (either because the feature is not available or because you pass `huge_tree=False`). Right now it fails silently, resulting in a wrong selector object.
If you have a better idea how to handle this let me know.

Reply:
@kmike @Langdi What would you think about making `LXML_SUPPORTS_HUGE_TREE` the default parameter value?

```python
def create_root_node(text, parser_cls, base_url=None, huge_tree=LXML_SUPPORTS_HUGE_TREE):
    # …
    parser_kwargs = {'recover': True, 'encoding': 'utf8'}
    if huge_tree:
        parser_kwargs['huge_tree'] = True
    parser = parser_cls(**parser_kwargs)
    # …
```

Reply:
Yes, that's probably a good idea because it has better backward compatibility.

Reply:
@Langdi Mind that switching `LXML_SUPPORTS_HUGE_TREE` as default value should also allow you to simplify the code significantly, specially if we want to let the code fail naturally if `huge_tree=True` is passed but not supported.